azurerm/internal/services/batch/resource_arm_batch_pool.go

```diff
@@ -372,6 +372,14 @@ func resourceArmBatchPool() *schema.Resource {
 					},
 				},
 			},
+			"metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
```

Comment:
Could we add an `NoEmptyStrings` validation here?

Reply:
done